earth2mip/inference_ensemble.py

```diff
@@ -258,7 +260,19 @@ def perturb(x, rank, batch_id, device):
             )
         if rank == 0 and batch_id == 0:  # first ens-member is deterministic
             noise[0, :, :, :, :] = 0
+        center = torch.tensor(
+            [channel_means[channel] for channel in model.in_channel_names],
+            device=x.device,
+        )
+        scale = torch.tensor(
+            [channel_stds[channel] for channel in model.in_channel_names],
+            device=x.device,
+        )
+        center_reshaped = center[None, None, :, None, None]
+        scale_reshaped = scale[None, None, :, None, None]
```

Comment:
To be a little more flexible I think you can get away with 
`scale_reshaped = scale[:, None, None]`

And let pytorch broadcast for you.